### PR TITLE
QP-1

### PR DESCRIPTION
**Pull request by jrrakh**
**QP-1** - Add ability for admin to create new directions (server-side). Done.

### CODE REVIEW

File: src/main/java/com/testportal/impl/web/controller/DirectionController.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+/**
+ * testportal
+ * com.testportal.impl.web.controller
+ * DirectionController - REST controller that provides working with directions.
+ * @see Direction
+ * @see DirectionService
+ * @author Roman_Rahozhyn
+ *         5/19/16
+ */
+@RestController
```
Comment (Ghost139) [inline src/main/java/com/testportal/impl/web/controller/DirectionController.java]: Are we really need REST here?

File: src/main/java/com/testportal/impl/dao/RoleDaoImpl.java [hunk 2]
```diff
@@ -17,7 +17,7 @@
 
     public static final String ADD_ROLE_TO_USER = "INSERT INTO ROLE VALUES (default, ?, ?)";
     //TODO Read SQL JOIN
-    public static final String GET_USER_ROLE = "SELECT DISTINCT role.role FROM role INNER JOIN user_role ON user_role.userId=?";
+    public static final String GET_USER_ROLE = "SELECT role FROM Role INNER JOIN (SELECT * FROM User_Role WHERE userId=?) AS User_Role ON Role.id=User_Role.roleId";
```
Comment (Ghost139) [inline src/main/java/com/testportal/impl/dao/RoleDaoImpl.java]: I'll think about this query :)

File: src/main/java/com/testportal/impl/web/controller/DirectionController.java (hunk 1 above)
Comment (jrrakh) [inline src/main/java/com/testportal/impl/web/controller/DirectionController.java]: I think we can use this in some admin panel via ajax.

File: src/main/java/com/testportal/impl/dao/RoleDaoImpl.java (hunk 2 above)
Comment (jrrakh) [inline src/main/java/com/testportal/impl/dao/RoleDaoImpl.java]: It will be good, if you have some ideas how to simplify this query.

**Pull request merged**